legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/mapping/MappingParseTreeWalker.java

```diff
@@ -156,35 +172,97 @@ else if (mappingElement instanceof AssociationMapping)
         return mapping;
     }
 
-    private MappingTest visitMappingTest(MappingParserGrammar.TestContext ctx, Mapping mapping)
+    private MappingTestSuite visitMappingTestSuite(MappingParserGrammar.MappingTestSuiteContext ctx, Mapping mapping)
+    {
+        MappingTestSuite mappingTestSuite = new MappingTestSuite();
+        mappingTestSuite.sourceInformation = this.walkerSourceInformation.getSourceInformation(ctx);
+        mappingTestSuite.id = PureGrammarParserUtility.fromIdentifier(ctx.identifier());
+
+        List<MappingParserGrammar.MappingTestDataContext> validatedDataCtx = PureGrammarParserUtility.validateRequiredListField(ctx.mappingTestData(), "data", mappingTestSuite.sourceInformation);
+        mappingTestSuite.storeTestDatas = ListIterate.collect(validatedDataCtx, this::visitMappingStoreTestData);
+        List<MappingParserGrammar.MappingTestContext> validatedTestCtx = PureGrammarParserUtility.validateRequiredListField(ctx.mappingTest(), "tests", mappingTestSuite.sourceInformation);
+        mappingTestSuite.tests = ListIterate.collect(validatedTestCtx, atomicTest -> this.visitAtomicTest(atomicTest, mapping));
+
+        return mappingTestSuite;
+    }
+
+    private StoreTestData visitMappingStoreTestData(MappingParserGrammar.MappingTestDataContext ctx)
+    {
+        StoreTestData testData = new StoreTestData();
+        testData.data = HelperEmbeddedDataGrammarParser.parseEmbeddedData(ctx.embeddedData(), this.walkerSourceInformation, this.parserContext.getPureGrammarParserExtensions());
+        this.createDummyBindingIfRequired(testData.data);
+        testData.store = ctx.qualifiedName().packagePath() == null && testData.data instanceof ModelStoreData ?
+                "ModelStore" :
+                PureGrammarParserUtility.fromQualifiedName(ctx.qualifiedName().packagePath().identifier(),ctx.qualifiedName().identifier());     //build store
+        return testData;
+    }
+
+    private void createDummyBindingIfRequired(EmbeddedData data)
+    {
+        if (data instanceof ModelStoreData)
+        {
+            ((ModelStoreData) data).instances.keySet().stream().forEach(key ->
```

Comment:
can we create just 1 binding with all elements? Also this might not be enough for nested elements as I believe compilation wpould fail for those

Do we compile this binding or just use as placeholder?

Reply:
we do compile these bindings, when building the graph, but we dont use it

Reply:
but since we dont use the binding at all, the process doesnt error out

Reply:
Do we have a test with 3 level graphFetchTree? If not lets add one